buildscripts/indexsymbols.py

```diff
@@ -0,0 +1,136 @@
+#               srctool.exe and pdbstr.exe
+#
+#############################################################################################
+
+import os.path
+import argparse
+import subprocess
+from subprocess import Popen, PIPE
+
+def write_index(index_filename, tag_version):
+    """This writes to a temporary index file for later use
+
+    For example
+
+    SRCSRV: ini ------------------------------------------------
+    VERSION=1
+    SRCSRV: variables ------------------------------------------
+    PATH=%var2%
+    SRCSRVTRG=%TARG%\%PDBVERSION%\%fnbksl%(%var2%)
+    SRCURL=https://raw.githubusercontent.com/Microsoft/msphpsql/%SRCVERSION%/source/%PATH%
+    SRCSRVCMD=powershell -Command "$r=New-Object -ComObject Msxml2.XMLHTTP; $r.open('GET', '%SRCURL%', $false); $r.send(); [io.file]::WriteAllBytes('%SRCSRVTRG%', $r.responseBody)"
+    SRCVERSION=v5.6.0
+    PDBVERSION=v5.6.0
+    For example
+    """
+    with open(index_filename, 'w') as f:
+        f.write('SRCSRV: ini ------------------------------------------------\n')
+        f.write('VERSION=1\n')
+        f.write('SRCSRV: variables ------------------------------------------\n')
+        f.write('PATH=%var2%\n')
+        f.write('SRCSRVTRG=%TARG%\%PDBVERSION%\%fnbksl%(%var2%)\n')
+        f.write('SRCURL=https://raw.githubusercontent.com/Microsoft/msphpsql/%SRCVERSION%/source/%PATH%\n')
+        f.write('SRCSRVCMD=powershell -Command ')
+        f.write('\"$r=New-Object -ComObject Msxml2.XMLHTTP; ')
+        f.write('$r.open(\'GET\', \'%SRCURL%\', $false); ')
+        f.write('$r.send(); [io.file]::WriteAllBytes(\'%SRCSRVTRG%\', $r.responseBody)\"\n')
+        f.write('SRCVERSION=' + tag_version + '\n')
+        f.write('PDBVERSION=' + tag_version + '\n')
+
+def append_source_filess(index_filename, source_files, driver):
+    """This appends the paths to different source files to the temporary index file
+
+    For example
+
+    SRCSRV: source files ---------------------------------------
+    c:\php-sdk\phpdev\vc15\x86\php-7.2.14-src\ext\pdo_sqlsrv\pdo_dbh.cpp*pdo_sqlsrv/pdo_dbh.cpp
+    c:\php-sdk\phpdev\vc15\x86\php-7.2.14-src\ext\pdo_sqlsrv\pdo_init.cpp*pdo_sqlsrv/pdo_init.cpp
+    ... ...
+    c:\php-sdk\phpdev\vc15\x86\php-7.2.14-src\ext\pdo_sqlsrv\shared\core_stream.cpp*shared/core_stream.cpp
+    c:\php-sdk\phpdev\vc15\x86\php-7.2.14-src\ext\pdo_sqlsrv\shared\core_util.cpp*shared/core_util.cpp
+    SRCSRV: end ------------------------------------------------
+    """
+    failed = False
+    with open(index_filename, 'a') as idx_file:
+        idx_file.write('SRCSRV: source files ---------------------------------------\n')
+        with open(source_files, 'r') as src_file:
+            for line in src_file:
+                pos = line.find('shared')
+                if (pos > 0):           # it's a nested folder, so it must be positive
+                    relative_path = line[pos:]
+                    src_line = line[:-1] + '*' + relative_path.replace('\\', '/')

```

Comment:
`line[:-1]` is excluding the newline at the end?

Reply:
yes